library/pathname/empty_spec.rb

```diff
@@ -0,0 +1,28 @@
+require File.expand_path('../../spec_helper', __dir__)
+require 'pathname'
+
+describe "Pathname#empty?" do
+  ruby_version_is "2.4" do
+    it 'returns true when file is not empty' do
+      (Pathname.new(__FILE__).empty?).should be_false
```

Comment:
Here I'm using the fact `__FILE__`  is this same file (and should not be empty), if you need to explicitly create a non empty file (as in `shared/file/zero.rb`), I can change it.

Reply:
This is perfectly fine :) One restriction is to try to open a spec file in write mode as sometimes specs are on a read-only disk or have read-only permissions. But this doesn't apply here.